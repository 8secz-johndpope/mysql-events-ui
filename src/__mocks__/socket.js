/* eslint-disable */

export default {
  connect: jest.fn()
}