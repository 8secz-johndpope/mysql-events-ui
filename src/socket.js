import io from 'socket.io-client'
import env from '@/env'
import store from '@/store'

let instance

export default {
  instance: () => instance,
  connect() {
    if (!instance) instance = io(env.socketServer)
    instance.on('connect', this.onConnect)
    instance.on('operationReceived', this.onOperationReceived)
    return instance
  },
  onConnect() {
    console.log('user connected')
  },
  onOperationReceived(data) {
    return {
      INSERT: () => store.dispatch('INSERT', data),
      UPDATE: () => store.dispatch('UPDATE', data),
      DELETE: () => store.dispatch('DELETE', data),
    }[data.type]()
  },
}