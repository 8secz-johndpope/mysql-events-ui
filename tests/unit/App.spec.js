jest.mock('@/socket')
import { shallowMount } from '@vue/test-utils'
import socket from '@/socket'
import App from '@/App'

describe('App', () => {
  const build = () => {
    const wrapper = shallowMount(App, {
      stubs: ['router-view']
    })

    return {
      wrapper,
    }
  }

  it('renders the component correctly', () => {
    // arranje
    const { wrapper } = build()

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('starts socket service', () => {
    // arranje
    build()

    // assert
    expect(socket.connect).toHaveBeenCalled()
  })
})