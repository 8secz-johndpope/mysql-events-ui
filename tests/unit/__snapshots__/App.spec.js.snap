// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component correctly 1`] = `
<div id="app">
  <router-view-stub></router-view-stub>
</div>
`;
